app/src/main/java/org/desarrolladorslp/teckersapp/ui/messages/MessageViewModel.kt

```diff
@@ -1,18 +1,48 @@
 package org.desarrolladorslp.teckersapp.ui.messages
 
-import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
-import androidx.lifecycle.Transformations
 import androidx.lifecycle.ViewModel
 
-class MessageViewModel : ViewModel() {
+import org.desarrolladorslp.teckersapp.model.Inbox
+import org.desarrolladorslp.teckersapp.model.MessageHeader
+import org.desarrolladorslp.teckersapp.service.APIEndpoint
+import org.desarrolladorslp.teckersapp.service.MessageService
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
 
-    private val _index = MutableLiveData<Int>()
-    val text: LiveData<String> = Transformations.map(_index) {
-        "Hello world from section: $it"
+class MessageViewModel() : ViewModel() {
+
+    private var messageService = APIEndpoint.instance()?.create(MessageService::class.java);
+
+    val _inbox= MutableLiveData<Inbox>()
+    val pageIndex=2
+    fun getmessages(priorityIndex : Int) :ArrayList<MessageHeader> {
```

Comment:
You should use the MutableLiveData object as I suggest in the corresponding class. This method is redundant.